src/Battlescape/ActionMenuState.cpp

```diff
@@ -300,7 +358,7 @@ void ActionMenuState::btnActionMenuItemClick(Action *action)
 				_game->popState();
 			}
 		}
-		else if (_action->type == BA_UNPRIME)
+		else if (_action->type == BA_UNPRIME && weapon->getBattleType() != BT_SCRIPTED)
```

Comment:
if prime and unprime are handled further below, shouldn't also BA_HIT be protected and handled below?

Reply:
BA_HIT should take the default code path and still work, that was the idea at least.
The switch statement below has a case for BA_HIT.